Descent3/menu.cpp

```diff
@@ -1398,6 +1455,76 @@ int DisplayLevelWarpDlg(int max_level) {
   hwnd.Destroy();
   return chosen_level;
 }
+
+// DisplayLevelSelectDlg
+// displays a list of levels associated with the selected mission
+int DisplayLevelSelectDlg(int max_level) {
+  newuiTiledWindow lvlsel;
+  newuiSheet *select_sheet;
+  newuiListBox *level_lb;
+  level_info lvinfo;
+  char level_name[100];
+  int chosen_level = 1, res;
+  char buffer[128];
+  int highest_allowed;
+
+  // Create the menu
+  lvlsel.Create(msnname, 0, 0, 448, 384);
+  select_sheet = lvlsel.GetSheet();
+  select_sheet->NewGroup(NULL, 10, 0);
+  level_lb = select_sheet->AddListBox(352, 256, UID_LVLB, UILB_NOSORT);
+  select_sheet->NewGroup(NULL, 160, 280, NEWUI_ALIGN_HORIZ);
+  select_sheet->AddButton(TXT_OK, UID_OK);
+  select_sheet->AddButton(TXT_CANCEL, UID_CANCEL);
+
+  // Loop through the levels
+  for (int level = 1; level <= max_level; ++level) {
+    // Check if msnname matches core mission names, if so use the arrays for level names
+    // Otherwise, try to get the level name from the level info
+    if (strcmp(msnname, "Descent 3: Retribution") == 0) {
```

Comment:
Maybe there can be more elegant way to determine on what campaign we choose?

Reply:
Sure, if we ever need the mission name in other files we can declare it earlier and make it a global option.